app/src/main/java/com/google/android/samples/socialite/ui/camera/CameraViewModel.kt

```diff
@@ -93,12 +98,42 @@ class CameraViewModel @Inject constructor(
         .setQualitySelector(QualitySelector.from(Quality.HIGHEST))
         .build()
 
-    private val videoCaptureUseCase = VideoCapture.Builder(recorder)
-        .build()
-
     private var currentRecording: Recording? = null
     private lateinit var recordingState: VideoRecordEvent
 
+    init {
+        val videoCaptureBuilder = VideoCapture.Builder(recorder)
+
+        viewModelScope.launch {
+            val hdrCameraInfo = getHdrCameraInfo()
+
+            if (hdrCameraInfo != null) {
+                Log.i(TAG, "Capturing HDR video")
+                videoCaptureBuilder.setDynamicRange(hdrCameraInfo)
+            }
+
+            videoCaptureUseCase = videoCaptureBuilder.build()
+        }
+    }
+
+    private suspend fun getHdrCameraInfo(): DynamicRange? {
+        var supportedHdrEncoding: DynamicRange? = null
+
+        cameraProviderManager.getCameraProvider().availableCameraInfos
+            .first { cameraInfo ->
+                val videoCapabilities = Recorder.getVideoCapabilities(cameraInfo)
+                val supportedDynamicRanges =
+                    videoCapabilities.supportedDynamicRanges
+
+                supportedHdrEncoding = supportedDynamicRanges.firstOrNull {
+                    it != DynamicRange.SDR // Ensure an HDR encoding is found
```

Comment:
Why don't you directly check HDR, not SDR?

Reply:
+1 to this, can we just check if it is HDR instead?

Reply:
i'm not sure, i copied the code snippet from https://android-developers.googleblog.com/2023/06/camerax-13-is-now-in-beta.html which has the snippet:
```
supportedHdrEncoding = supportedDynamicRanges.firstOrNull {
            it != DynamicRange.SDR  // Ensure an HDR encoding is found
        }
```

@donovanfm any idea why we are checking for NOT SDR instead of checking for HDR directly?